skll/config.py

```diff
@@ -807,9 +805,9 @@ def _parse_config_file(config_path, log_level=logging.INFO):
     if task in ['cross_validate', 'train', 'learning_curve'] and test_path:
         raise ValueError('The test set should not be set when task is '
                          '{}.'.format(task))
-    if task in ['train', 'predict'] and results_path:
+    if task in ['train', 'predict'] and results_path and not do_grid_search:
         raise ValueError('The results path should not be set when task is '
-                         '{}.'.format(task))
+                         '{} and "grid_search" is set to False.'.format(task))
```

Comment:
Does it make sense to fail out from `predict` too? That task doesn’t do anything with grid search and has no results.

Reply:
It looks like one can set grid search to True for `predict` as well as `train` and it will happen the same way in both cases. That is, if `train`, the model that results will use the best parameters. If `predict`, the model that gets trained and is used to predict on the test set will use the best parameters found during grid search. So, in the latter case, the results that are provided will be exactly the same as the former case. Is this incorrect?

Reply:
Ah, of course. I forgot that `predict` will train a model. I thought it always loads an existing model. Hmm, actually what happens in this PR in the case where we are reusing an already trained model, e.g., by specifying `-k` at the command line?

Reply:
Good point. I'll take a look at that case. I'm not sure what will happen.

Reply:
In cases where a results path is set and grid search is on but then grid search is actually not done, results that are reported will basically have grid score/grid search CV results set to `null`, so the results will not be very informative (though seeing `null` values will tell the user grid search was not done, at least). There are two situations where this will happen:
1. When you're reusing a model. In this case, grid search has already been done (presumably) and there's nothing to do. The results that will be generated will overwrite the previous results file, however, which is not great if you wanted to use those results, but that's a pretty strange edge case. Perhaps a check could be introduced to not overwrite results if they exist and you're reusing a model that exists. That could be complicated.
2. When you're using an empty parameter grid, in which case there's no grid search to do. This might seem unlikely, but it's possible that the user specifies a learner whose default parameter grid is empty, like `LinearRegression`. In that case, the results will also be empty in terms of grid search-related attributes.

In both cases, I would argue that the behavior is expected. If you're using `-k`, you should know that grid search won't be done and setting grid search to true/specifying a results path is pointless. In the case where grid search won't be done due to there being no grid to search, the results related to the grid search will be accurate at the very least.

Reply:
Thanks for investigating. I think this is fine but we will definitely need to update documentation to briefly mention some of these things. Please update the documentation issue to include this.